crates/uv-distribution/src/distribution_database.rs

```diff
@@ -45,6 +45,14 @@ pub struct DistributionDatabase<'a, Context: BuildContext + Send + Sync> {
     builder: SourceDistCachedBuilder<'a, Context>,
 }
 
+fn handle_response_errors(err: reqwest::Error) -> io::Error {
+    if err.is_timeout() {
+        io::Error::new(io::ErrorKind::TimedOut, "Failed to download distribution due to network timeout. Try increasing UV_HTTP_TIMEOUT.")
```

Comment:
It looks like Zanie suggested `HTTP_TIMEOUT`. Do you mind changing to `HTTP_TIMEOUT`, and displaying the current value too? This is great!

Reply:
Good point, I decided to use `UV_HTTP_TIMEOUT` for consitency with `registry_client.rs` first check (which seems to prefer `UV_HTTP_TIMEOUT` and `HTTP_TIMEOUT` last). I could also update that `warn_user_once` as part of this PR if we want to be fully consistent.

```rust
            let timeout = env::var("UV_HTTP_TIMEOUT").or_else(|_| env::var("UV_REQUEST_TIMEOUT")).or_else(|_| env::var("HTTP_TIMEOUT")).and_then(|value| {
                value.parse::<u64>()
                    .or_else(|_| {
                        // On parse error, warn and  use the default timeout
                        warn_user_once!("Ignoring invalid value from environment for UV_REQUEST_TIMEOUT. Expected integer number of seconds, got \"{value}\".");
                        Ok(default_timeout)
                    })
            }).unwrap_or(default_timeout);
```

Reply:
I think `UV_HTTP_TIMEOUT` is good, it seems good not to recommend changing a value that other clients will respect unless you need to.

Reply:
Can we say how long the timeout is currently though?

Reply:
Added in 65a5c3542357aa1eb69950cc0b4baa316ae82ab2 and 783a4196d09f3ba10a0294529a85528c621ee8fd. I didn't see reqwest allowing us to look at config.timeout, so I ended up exposing our timeout to the RegistryClient struct instead.